### PR TITLE
Issue/5041 fix documentation build issues

### LINKED ISSUE #5041: Fix documentation build issues
A build of the documentation of inmanta-core shows the following message:

```
<sphinxcontrib.inmanta.config>:1: WARNING: inmanta.config:option reference target not found: server.available-versions-to-keep
```
and 
```
install/install-server/install-server.inc:71: ERROR: Unknown target name: "web-console github page<https://github.com/inmanta/web-console/packages/>".
install/install-server/install-server.inc:117: ERROR: Unknown target name: "web-console github page<https://github.com/inmanta/web-console/packages/>".
```
and
```
inmanta-core/docs/platform_developers/features.rst:7: ERROR: Error in "code-block" directive:
maximum 1 argument(s) allowed, 5 supplied.

.. code-block:: yaml
    slices:
        slice_name:
            feature_name: bool
```
and
```
inmanta-core/docs/adr/0000-logging-warnings-using-the-python-warnings-or-logging-module.md: WARNING: document isn't included in any toctree
inmanta-core/docs/adr/template.md: WARNING: document isn't included in any toctree
```
and
```
/home/arnaud/irt/output/documentation_base_repo/docs/moduleguides/graph/index.rst:39: ERROR: Error in "code-block" directive:
maximum 1 argument(s) allowed, 3 supplied.

.. code-block:: bash
  graph::Graph(name="my_graph", config=std::source("/files_and_hosts.g"))
```

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

Fix errors appearing when using the make docs command.

closes #5041

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Comment by Hugo-Inmanta** [comment]
> @Hugo-Inmanta this one as no reviewers assigned?

I think there were some new errors/warnings that have popped up, I figured I might as well fix them with this PR. I'll get back to this asap

**Comment by Hugo-Inmanta** [comment]
There are still a couple of remaining warnings that I don't know how to fix e.g.:
```
/home/hugo/work/inmanta/github-repos/inmanta-core/src/inmanta/protocol/methods.py:docstring of inmanta.protocol.methods.clear_environment:3: WARNING: Field list ends without a blank line; unexpected unindent.
/home/hugo/work/inmanta/github-repos/inmanta-core/src/inmanta/protocol/methods.py:docstring of inmanta.protocol.methods.clear_environment:5: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/hugo/work/inmanta/github-repos/inmanta-core/src/inmanta/protocol/methods_v2.py:docstring of inmanta.protocol.methods_v2.agent_action:8: WARNING: Inline strong start-string without end-string.
```

File: src/inmanta/module.py [hunk 1]
```diff
@@ -1518,40 +1518,41 @@ class ProjectMetadata(Metadata, MetadataFieldRequires):
...
-        The repository should be `PEP 503 <https://www.python.org/dev/peps/pep-0503/>`_ (the simple repository API) compliant.
+        * git: When the type is set to git, the url field should contain a template of the Git repo URL. Inmanta creates the
+            git repo url by formatting {} or {0} with the name of the module. If no formatter is present it appends the name
+            of the module to the URL.
 
-      The old syntax, which only defines a Git URL per list entry is maintained for backward compatibility.
+        * package: When the type is set to package, the URL field should contain the URL of the Python package repository.
+            The repository should be `PEP 503 <https://www.python.org/dev/peps/pep-0503/>`_ (the simple repository API)
+            compliant.
+
```
Comment (arnaudsjs) [inline src/inmanta/module.py:1535]: This doesn't render well in the documentation

Screenshot from 2023-01-17 13-23-21

File: src/inmanta/protocol/methods_v2.py [hunk 2]
```diff
@@ -1063,24 +1076,26 @@ def get_parameters(
...
     :param filter: Filter the list of returned parameters.
-                The following options are available:
-                name: filter by the name of the parameter
-                source: filter by the source of the parameter
-                updated: filter by the updated time of the parameter
+
+        The following options are available:
+            name: filter by the name of the parameter
+            source: filter by the source of the parameter
+            updated: filter by the updated time of the parameter
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:1095]: In the API documentation does doesn't render as an itemization. But I am not sure whether we can do something about this.
Screenshot from 2023-01-17 13-28-51

File: src/inmanta/protocol/methods_v2.py [hunk 3]
```diff
@@ -764,42 +765,49 @@ def resource_logs(
...
+            returned.
+
+            message: filter by the content of the log messages. Partial matches are allowed. (case-insensitive)
+
+            minimal_log_level: filter by the log level of the log messages. The filter specifies the minimal level,
+            so messages with either this level, or a higher severity level are going to be included in the result.
+
+            For example, for `filter.minimal_log_level=INFO`, the log messages with level `INFO, WARNING, ERROR, CRITICAL`
+            all match the query.
+
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:806]: This also renders rather weird in the API documentation. I am also not sure if we can do something about it.
Screenshot from 2023-01-17 13-37-19

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/protocol/methods_v2.py (hunk 3 above)
Comment (Hugo-Inmanta) [reply src/inmanta/protocol/methods_v2.py]: I added some bullet points and the programmatic API reference now looks like this:

image

I haven't check how the API doc itself renders, I don't know/remember how to generate it

Comment (arnaudsjs) [reply src/inmanta/protocol/methods_v2.py]: Nice!

File: src/inmanta/protocol/methods_v2.py (hunk 2 above)
Comment (Hugo-Inmanta) [reply src/inmanta/protocol/methods_v2.py]: Now looks like this:
image

File: src/inmanta/module.py (hunk 1 above)
Comment (Hugo-Inmanta) [reply src/inmanta/module.py]: image

File: src/inmanta/protocol/methods_v2.py (hunk 3 above)
Comment (arnaudsjs) [reply src/inmanta/protocol/methods_v2.py]: You can render the documentation using `make docs`.

Comment (Hugo-Inmanta) [reply src/inmanta/protocol/methods_v2.py]: Yes, that's what I used and then checkout the output in docs/build/html/index.html but it doesn't render the same on my end so I thought the API documentation you were mentioning was something else.

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**